include/aspect/material_model/viscoelastic.h

```diff
@@ -0,0 +1,273 @@
+/*
+  Copyright (C) 2014 - 2017 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#ifndef _aspect_material_model_viscoelastic_h
+#define _aspect_material_model_viscoelastic_h
+
+#include <aspect/material_model/interface.h>
+#include <aspect/simulator_access.h>
+#include <deal.II/base/parameter_handler.h>
+
+namespace aspect
+{
+  namespace MaterialModel
+  {
+    using namespace dealii;
+
+    /**
+    * Additional output fields for the elastic shear modulus to be added to
+    * the MaterialModel::MaterialModelOutputs structure and filled in the
+    * MaterialModel::Interface::evaluate() function.
+    */
+    template <int dim>
+    class ElasticAdditionalOutputs : public NamedAdditionalMaterialOutputs<dim>
+    {
+      public:
+        ElasticAdditionalOutputs(const unsigned int n_points);
+
+        virtual std::vector<double> get_nth_output(const unsigned int idx) const;
+
+        /**
+         * Elastic shear moduli at the evaluation points passed to
+         * the instance of MaterialModel::Interface::evaluate() that fills
+         * the current object.
+         */
+        std::vector<double> elastic_shear_moduli;
+    };
+
+    /**
+     * A material model which is intended for simple linear viscoelastic
+     * problems with multiple compositional fields.
+     *
+     * The viscoelastic constitutive relationship and implementation follows
+     * a method commonly used in the Geodynamics community, where only the
+     * elastic shear strength of rocks (e.g., shear modulus) is considered.
+     * The material model directly follows the work flow and terminology
+     * of Moresi et al. 2003 (J. Comp. Phys., v. 184, p. 476-497) equations
+     * 23-32, which is commonly referred to in proceeding geodynamic
+     * publications. However, a notable difference between this material
+     * model and that of previous work is the use of compositional fields,
+     * rather than tracers, to track and advect the stress tensor. The
+     * material model will be updated when an option to track and advect
```

Comment:
Instead of writing this, put it into a TODO, as in `TODO: Add option to track and advect stresses with tracers.` Makes it easier to find.